libs/vkd3d/resource.c

```diff
@@ -5387,6 +5403,7 @@ static HRESULT d3d12_descriptor_heap_init(struct d3d12_descriptor_heap *descript
     descriptor_heap->refcount = 1;
     descriptor_heap->device = device;
     descriptor_heap->desc = *desc;
+    descriptor_heap->null_descriptor_template.has_mutable_descriptors = device->vk_info.VALVE_mutable_descriptor_type;
```

Comment:
Should probably be moved to add_null_descriptor_template path where stuff is set up once.